sync/sync.go

```diff
@@ -346,6 +347,18 @@ Register:
 		return fmt.Errorf("failed to set rebuilding: true, error: %s", err.Error())
 	}
 
+	ok, err := t.isRevisionCountAndChainSame(replicaAddress, fromClient, toClient)
+	if err != nil {
+		return err
+	}
+
+	if ok {
+		if err := toClient.SetRebuilding(false); err != nil {
+			return fmt.Errorf("Failed to set rebuilding: true, error: %v", err)
+		}
+		return nil
+	}
+
 	logrus.Infof("PrepareRebuild %v", replicaAddress)
 	output, err := t.client.PrepareRebuild(rest.EncodeID(replicaAddress))
```

Comment:
how about doing PrepareRebuild as well, as it does sync of head meta files?

Reply:
No @vishnuitta , PrepareRebuild only returns the chain[0:] but we compare content of the chain whether it starts with prefix `volume-head`, we ignore the sync of head files (both head and meta).